design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+
+* If `--include-cluster-scope-resources` is not present, it means no additional cluster resource is included per resource type.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources` have interact with `--include-resources` and `--exclude-resources`. The rule is specified in [IncludedResources and ExcludedResources change](#includedresources-and-excludedresources-change)
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` have interact with `--include-cluster-resources` parameter. The rule is specified in [IncludeClusterResources change](#includeclusterresources-change)
+
+* If the backup or restore contains no resource, validation failure should be returned.
+
+### Using scenarios:
+#### no namespaced resources + no cluster resources
+This is not allowed. Backup or restore cannot contain no resource.
+
+#### no namespaced resources + some cluster resources
+The following commands all mean backup no namespaced resources and some cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+#### no namespaced resources + all cluster resources
+The following commands all mean backup no namespaced resources and all cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=*
+```
+
+#### some namespaced resources + no cluster resources
+This means backup all resources in namespaces default and kube-system, and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
```

Comment:
This one should be fine as-is. Since we're not using any of the new params, the `--include-resources="*"` implied by an empty param should apply.